### DISCUSSION TITLE
mosquitto

### DISCUSSION THREAD

**Discussion by nerohmot** [Semi-ATE/Semi-ATE#135] [Q&A]
So we will install `mosquitto` on the MiniSCT as part of the "OS installation" with `apt-get install ...`
Question now is how do we configure/enable/disable the mosquitto ?

**Comment by Zlin526F** [comment]
This will be ok, if Semi-ATE running on a MiniSCT, but if it's running on another system like windows together with NI-Instruments, an other solution without mosquitto would be helpful.  Installing mosquitto on windows is sometimes difficult, you need admin rights, and have to restart your computer. 
What do you think about python [HBMQTT](https://hbmqtt.readthedocs.io/en/latest/).

**Reply by AchimLott** [reply]
Of course, Chris, installing Mosquitto on a windows system requires admin rights - that's what the admin role is supposed for ;-)
The installation itself isn't complex, if you know what you do and can certainly be automated by our IT guys.

**Reply by Zlin526F** [reply]
I know the pain of installing other users.  All had problems.
You can  also install Mosquitto without admin rights, or you can login as admin, but run it without admin rights.  After installation no error occur, and so you think everything is ok. But if you restart your computer, Mosquitto will not starting. No info/warning/error occur. Also in Semi-ATE no error occur, but it isn't running correctly. And than the user ask you, why isn't semi-ate running.... and you search and search.. I think the best solution is to install the whole environment from semi-ate without admin rights. 
Keep it simple, one installation script for the whole installation. Only in this way we will not get any queries from users.

**Comment by nerohmot** [comment]
- whatever we use, it should be in the form of a `conda` package (or at least a pip package) so we can add it to our development environment.|
- if Semi-ATE doesn't find a broker to connect to, it shoud pop up a warning

**Reply by awinia-github** [reply]
I agree to show some warning in case that the broker cannot be found. This will reduce bug hunting time. I am unsure about the conda package and also about HBMQTT.

1. Conda package generating and package maintaining may lead to some extra work, depending on which mqtt-broker we are using in the end.
2. The supported protocols are different. I.e. HBMQTT supports protocol 3.11 and Mosquitto supports protocol 5.0. Also the Github Stars are not in favour of HBMQTT.
